test/Array/array_findlast.js

```diff
@@ -0,0 +1,423 @@
+
+            called = 0;
+            result = arr.findLast(function(val) {
+                called++;
+                return val === "Shoes";
+            });
+
+            assert.areEqual(3, called, "predicate was called three times");
+            assert.areEqual("Shoes", result);
+
+            result = arr.findLast(function(val) {
+                return "string";
+            });
+            assert.areEqual("Bike", result, "coerced string");
+
+            result = arr.findLast(function(val) {
+                return {};
+            });
+            assert.areEqual("Bike", result, "coerced object");
+
+            result = arr.findLast(function(val) {
+                return Symbol("");
+            });
+            assert.areEqual("Bike", result, "coerced Symbol");
+
+            result = arr.findLast(function(val) {
+                return 1;
+            });
+            assert.areEqual("Bike", result, "coerced number");
+
+            result = arr.findLast(function(val) {
+                return -1;
+            });
+            assert.areEqual("Bike", result, "coerced negative number");
+        }
+    },
+    {
+        name : "return undefined if predicate returns false value",
+        body : function ()
+        {
+            var arr = ["Shoes", "Car", "Bike"];
+            var called = 0;
+
+            var result = arr.findLast(function(val) {
+                called++;
+                return false;
+            });
+
+            assert.areEqual(3, called, "predicate was called three times");
+            assert.areEqual(undefined, result);
+
+            result = arr.findLast(function(val) {
+                return "";
+            });
+            assert.areEqual(undefined, result, "coerced string");
+
+            result = arr.findLast(function(val) {
+                return undefined;
+            });
+            assert.areEqual(undefined, result, "coerced undefined");
+
+            result = arr.findLast(function(val) {
+                return null;
+            });
+            assert.areEqual(undefined, result, "coerced null");
+
+            result = arr.findLast(function(val) {
+                return 0;
+            });
+            assert.areEqual(undefined, result, "coerced 0");
+
+            result = arr.findLast(function(val) {
+                return NaN;
+            });
+            assert.areEqual(undefined, result, "coerced NaN");
+        }
+    },
+    {
+        name : "return last item",
+        body : function ()
+        {
+            var arr = [ { value: "Shoes" }, { value: "Car" }, { value: "Shoes" }, { value: "Bike" } ];
+            var called = 0;
+
+            var result = arr.findLast(function (val) {
+                called++;
+                return val.value === "Shoes";
+            });
+
+            assert.areEqual("Shoes", result.value);
+            assert.areEqual(arr[2], result);
+            assert.areEqual(2, called);
+        }
+    }
+]
+

```

Comment:
Because of the awkwardness around match lengths should have test examples with length limits - both here and in findIndex - try a length of 2^32 and a length of 0 and a length of 1;

Reply:
I'm not sure 100% get your point but I have try to cover that.